docs/website/docs/user-guides/v3-migration-guide.md

```diff
@@ -0,0 +1,145 @@
+#### Step 1 
+`cd` into the component directory, and delete the component from the Kubernetes cluster:
+```shell
+odo delete
+```
+#### Step 2
+Download and [install odo v3](../overview/installation.md).
+
+#### Step 3
+Run [`odo dev`](../command-reference/dev.md) to start developing your application using `odo` v3.
+
+#### Step 4
+Run `odo list` to see a list of components that are running on the cluster and what version of `odo` they are running.
+
+### Where are `odo push` and `odo watch` commands?
+`odo push` and `odo watch` have been replaced in v3 by a single command - `odo dev`. 
+
+In v2, if you wanted to automatically sync the code on developer system with the application running on a Kubernetes
+cluster, you had to perform two steps - first `odo push` to start the application on the cluster, and second `odo watch`
+to automatically sync the code. In v3, `odo dev` performs both actions with a single command.
+
+`odo dev` is not _just_ a replacement for these two commands. It's also different in behaviour in that, it's a 
+long-running process that's going to block the terminal. Hitting `Ctrl+c` will stop the process and cleanup the 
+component from the cluster. In v2, you had to use `odo delete`/`odo component delete` to delete inner loop resources 
+of the component from the cluster.
+
+### What happened to Ingress/Route?
+If you have used `odo` v2, you must have used Ingress (on Kubernetes) or Route (on OpenShift) to access the 
+application that was pushed to the cluster using `odo push`. `odo` v3 no longer creates an Ingress or a Route. Instead,
+it uses port-forwarding.
+
+When running `odo dev`, `odo` forwards a port on the development system to the port on the container cluster allowing 
+you remote access to your deployed application. It also prints the information when the application has started on the
+cluster:
+```shell
+$ odo dev
+...
+...
+-  Forwarding from 127.0.0.1:40001 -> 8080
+```
+This indicates that the port 40001 on the development system has been forwarded to port 8080 of the application 
+represented by the current `odo` component.
+
+:::info NOTE
+`odo` no longer supports creation of Ingress / Route out of the box. The `odo url` set of commands no longer exist 
+in v3.
+:::
+
+### Changes to the way component debugging works
+In `odo` v2, `odo push --debug` was used to run a component in debug mode. To setup port forwarding to the component's
+debug port, you had to run `odo debug port-forward`.
+
+In `odo` v3, you need to specify the debug port in the `devfile.yaml` as an endpoint, and run `odo dev --debug` to 
+start the component in debug mode. For example, a `container` component in the devfile should look like below, where 
+port 3000 is the application port and 5858 is the debug port:
+
+```yaml
+- name: runtime
+  container:
+    image: registry.access.redhat.com/ubi8/nodejs-12:1-36
+    memoryLimit: 1024Mi
+    endpoints:
+    - name: "3000-tcp"
+      targetPort: 3000
+    - name: "5858-tcp"
```

Comment:
Could you call it `debug` instead? This is the convention we will be trying to use for Debug endpoints. See https://github.com/redhat-developer/odo/issues/5988#issuecomment-1239497488 and https://github.com/redhat-developer/odo/issues/6036
Setting also the exposure to "none", otherwise the endpoint will be "public" by default.

```suggestion
    - name: debug
      exposure: none
```

Reply:
Done.